jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -973,8 +973,12 @@ def _update_stretch_histogram(self, msg={}):
                 x_max = x_limits.max()
                 y_min = max(y_limits.min(), 0)
                 y_max = y_limits.max()
-
                 arr = comp.data[y_min:y_max, x_min:x_max]
+                if self.config == "imviz":
+                    # Downsample input data to about 400px (as per compass.vue) for performance.

```

Comment:
It shouldn't matter that this can change the aspect ratio of the image, right? It's just being used for statistics?

Reply:
The data is flattened anyway (using `ravel()`) for stats, so no, I don't think aspect ratio is important here.

Reply:
Makes sense. Another thought: this is basically making the maximum image size roughly 600x600 (which is when round(size/400) becomes greater than 1). If you have a 1000 x 50 array or something like that, the y dimension will be downsampled even though the image area is less than the maximum 600x600. Should the constraint be on area rather than an individual dimension size?

Reply:
> Should the constraint be on area rather than an individual dimension size?

I am not sure. In the case of 1000 x 50 (`nx` x `ny`), it would become 500 x 50. This means X is sampled every other column, but Y is fully sampled. So if this is a star, imagine taking every other vertical strip off it and then doing stats on them.

Does it really affect the final histogram or vmin/vmax in a meaningful way? Not sure. If you have a science image like this, you can see how the histogram and vmin/vmax behave with and without this patch.

Also, if Tom R is going to do it properly upstream soon, then maybe we shouldn't waste too much time on this. I don't remember if he gave a concrete timeline this morning or not.

Reply:
Good questions though! Do you want to investigate the science image case with 1000 x 50? If not, I can try to do it tomorrow.

Reply:
Okay, so I did a quick check and posting my results here. You can decide whether these are good or bad.

I make a 1000 x 50 using one of the data in ImvizDitheredExample notebook and displays it:

```python
from astropy import units as u
from astropy.io import fits
from astropy.nddata import NDData
from astropy.utils.data import download_file
from astropy.wcs import WCS

from jdaviz import Imviz

imviz = Imviz()
imviz.show()

acs_47tuc_1 = download_file(
    'https://mast.stsci.edu/api/v0.1/Download/file?uri=mast:HST/product/jbqf03gjq_flc.fits', cache=True)
pf = fits.open(acs_47tuc_1)
a = NDData(pf[1].data[:50, :1000] * u.electron, wcs=WCS(pf[1].header, pf))
imviz.load_data(a, data_label="long_image")
```

| On main | This PR |
| --- | --- |
| ![full_hist_long_im](https://github.com/spacetelescope/jdaviz/assets/2090236/a8073d50-426e-4c12-8990-f8639fefbea4) | ![subsampled_hist_long_im](https://github.com/spacetelescope/jdaviz/assets/2090236/fd676843-fea7-47f8-9feb-a875853feb47) |
| ![full_vminmax_long_im](https://github.com/spacetelescope/jdaviz/assets/2090236/0901cec3-9e12-417b-9d55-48cb49418e53) | ![subsampled_vminmax_long_im](https://github.com/spacetelescope/jdaviz/assets/2090236/ca2560e4-c9f4-4f00-8089-3f3261085bcf) |

p.s. The aspect ratio thingy does affect the Compass visualization though given we have plan to refactor that to use something completely different, I am not too worried about it.

https://github.com/spacetelescope/jdaviz/blob/4b9a1146ea5c94dc68c4728526ed376c14939556/jdaviz/configs/imviz/plugins/viewers.py#L257

Reply:
On the test image it doesn't really impact the final histogram, and if this is being changed upstream then i agree not to waste time.